### PR TITLE
JBIDE-27769: quarkus wizard eats text

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@jeffmaury corrected several things and added this in a commit to your branch:
* wizard should not expand hoiz & vertically (added preferred sizes to each page)
* explanatory labels on top of each page should wrap if wizard dialog is resized
* list & tables with quarkus extensions should have scrollbars if wizard dialog is resized
Screen Recording 2021-05-28 at 12 10 37(2)

File: plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java [hunk 1]
```diff
@@ -36,6 +36,14 @@ public void run(String[] params, ICheatSheetManager manager) {
 		createProjectWizard.init(workbench, null);
 		WizardDialog dialog = new WizardDialog(window.getShell(), createProjectWizard);
 		dialog.create();
+//		dialog.setPageSize(200, 200);
```
Comment (sbouchet) [inline plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java:39]: please remove the comments.

**Review by sbouchet — APPROVED** [review]
This new patch fixes the second page's weird rendering. good job!

File: plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java (hunk 1 above)
Comment (adietish) [reply plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java]: done

**Comment by adietish** [comment]
@jeffmaury leaving you the final word and decision on the merge

File: plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java [hunk 2]
```diff
@@ -36,14 +36,6 @@ public void run(String[] params, ICheatSheetManager manager) {
 		createProjectWizard.init(workbench, null);
 		WizardDialog dialog = new WizardDialog(window.getShell(), createProjectWizard);
 		dialog.create();
-//		dialog.setPageSize(200, 200);
-//		dialog.getShell().computeSize(200, 200);
-//		dialog.getShell().layout(true);
-//		dialog.getShell().setSize(200,  200);
-//		dialog.getShell().pack();
-		dialog.getShell().pack();
```
Comment (sbouchet) [inline plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java]: seems that you removed more than expected...

**Review by adietish — APPROVED** [review]
added my PR on top. Works for me now.

File: plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java (hunk 2 above)
Comment (adietish) [reply plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java]: @sbouchet good catch, but this is on purppose. This was a leftover of when I tried to set the wizard shell size. Didn't see that being a "new wizard", this action isn't used when you invoke the wizard via ctrl + n > quarkus

Comment (sbouchet) [reply plugins/org.jboss.tools.quarkus.cheatsheet/src/org/jboss/tools/quarkus/cheatsheet/OpenCreateProjectWizardAction.java]: ok, thanks for the explanation. I tested again and still looks good on my side.

**Comment by jeffmaury** [comment]
I am still having the wizard taking all vertical space. Seems to be related to list in the extensions page that is eating that space. Hard to review because of formatting

**Comment by sbouchet** [comment]
> I am still having the wizard taking all vertical space. Seems to be related to list in the extensions page that is eating that space. Hard to review because of formatting

yes, me too but now the horizontal space is correctly rendered and the text is fully visible. The issue was to fix the horizontal space, notthe vertical one.

**Comment by jeffmaury** [comment]
@sbouchet @adietish I added a vertical hint for the tables to fix the vertical issue. if you can test it on Linux/MacOS before EOD, that would be cool

**Comment by sbouchet** [comment]
tested OK on Linux.

**Comment by adietish** [comment]
> I am still having the wizard taking all vertical space. Seems to be related to list in the extensions page that is eating that space. Hard to review because of formatting

sorry about the reformatting. I had to do it so that I could identify and grasp all all the layout instructions.

**Pull request merged**